backend/data.go

```diff
@@ -37,11 +41,45 @@ func (fn QueryDataHandlerFunc) QueryData(ctx context.Context, req *QueryDataRequ
 // QueryDataRequest contains a single request which contains multiple queries.
 // It is the input type for a QueryData call.
 type QueryDataRequest struct {
+	ForwardHTTPHeaders
```

Comment:
Why do we need this `ForwardHTTPHeaders` field declared in this struct? I cannot find it used anywhere, so this will always be `nil`. `QueryDataRequest` already implements `ForwardHTTPHeaders` itself since it implements all its methods. There's the same thing also in `CheckHealthRequest` and `CallResourceRequest`

Reply:
Sorry, it's a way to say that this struct implements the `ForwardHTTPHeaders` interface. I removed this now and added a check that verifies the requests implements this interface instead.

Reply:
Gotcha. Looks good now!